File: 804-unique-morse-code-words/804-unique-morse-code-words.cpp
class Solution {
public:
    int uniqueMorseRepresentations(vector<string>& words) {
        std::array<string, 26> tab = {".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."};
        unordered_set<string> transfo;
        for (auto &w : words) {
            vector<string> t;
            for (auto &c : w) {
                t.push_back(tab[(int)c - (int)'a']);
            }
            std::string s;
            for (auto & t_s : t) {
                s += t_s;
            }
            transfo.insert(s);
        }
        return transfo.size();
    }
};

File: 804-unique-morse-code-words/804-unique-morse-code-words.py
class Solution:
    def uniqueMorseRepresentations(self, words: List[str]) -> int:
        tab = {}
        for i, code in enumerate([".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."]):
            tab[chr(ord("a") + i)] = code
        transfo = set()
        for w in words:
            t = []
            for c in w:
                t.append(tab[c])
            transfo.add("".join(t))
        return len(transfo)